manager/controllers/app/blueprint_controller.go

```diff
@@ -116,15 +116,10 @@ func (r *BlueprintReconciler) reconcileFinalizers(blueprint *app.Blueprint) (ctr
 	return ctrl.Result{}, nil
 }
 
-func getReleaseName(blueprintName string, step app.FlowStep) string {
-	fullName := blueprintName + "-" + step.Name
-	return utils.HelmConformName(fullName)
-}
-
 func (r *BlueprintReconciler) deleteExternalResources(blueprint *app.Blueprint) error {
 	errs := make([]string, 0)
 	for _, step := range blueprint.Spec.Flow.Steps {
-		releaseName := getReleaseName(blueprint.Name, step)
+		releaseName := utils.GetReleaseName(blueprint.Labels[app.ApplicationNameLabel], blueprint.Labels[app.ApplicationNamespaceLabel], step)
```

Comment:
When the application controller creates the plotter the name will be `appName + "-" + appNamespace`. The plotter will create a blueprint per cluster and will name it the same. Thus a blueprint name should already have app name and app namespace in the name. Why do we need to pass this here separately?

Reply:
appName1 = "a", NS1 = "a-a"
appName2 = "a-a", NS2 = "a"

We have appName1-NS1 == appName2-NS2

I wanted to avoid such cases using hash. But we can decide to handle it together in the future.

Reply:
You can't have two blueprints with the same name.

Reply:
Changed it to be:
```golang
// Generate release name from step name
func GetReleaseNameByStepName(applicationName string, namespace string, stepName string) string {
	fullName := applicationName + "-" + namespace + "-" + stepName
	return HelmConformName(fullName)
}

// Generate fqdn for a module
func GenerateModuleEndpointFQDN(releaseName string, blueprintNamespace string) string {
	return releaseName + "." + blueprintNamespace + ".svc.cluster.local"
}
```